charms/kserve-controller/src/charm.py

```diff
@@ -46,6 +48,25 @@
 
 CONFIG_FILES = ["src/templates/configmap_manifests.yaml.j2"]
 CONTAINER_CERTS_DEST = "/tmp/k8s-webhook-server/serving-certs/"
+DEFAULT_IMAGES = {
+    "configmap__agent": "kserve/agent:v0.10.0",
+    "configmap__batcher": "kserve/agent:v0.10.0",
+    "configmap__explainers__alibi": "kserve/alibi-explainer:latest",
+    "configmap__explainers__aix": "kserve/aix-explainer:latest",
+    "configmap__explainers__art": "kserve/art-explainer:latest",
+    "configmap__logger": "kserve/agent:v0.10.0",
+    "configmap__router": "kserve/router:v0.10.0",
+    "configmap__storageInitializer": "kserve/storage-initializer:v0.10.0",
+    "serving_runtimes__lgbserver": "kserve/lgbserver:v0.10.0",
+    "serving_runtimes__kserve_mlserver": "docker.io/seldonio/mlserver:1.0.0",
+    "serving_runtimes__paddleserver": "kserve/paddleserver:v0.10.0",
+    "serving_runtimes__pmmlserver": "kserve/pmmlserver:v0.10.0",
+    "serving_runtimes__sklearnserver": "kserve/sklearnserver:v0.10.0",
+    "serving_runtimes__tensorflow_serving": "tensorflow/serving:2.6.2",
+    "serving_runtimes__torchserve": "pytorch/torchserve-kfs:0.7.0",
+    "serving_runtimes__tritonserver": "nvcr.io/nvidia/tritonserver:21.09-py3",
+    "serving_runtimes__xgbserver": "kserve/xgbserver:v0.10.0",
+}
```

Comment:
Will we always have an opinion on what the "default" images here should be, or do we just want to use kserve's defaults unless the user overrides them?  If we use kserve's defaults, then we don't need to maintain this list here.  

My guess is that, although this list right now has upstream kserve images, soon we'll populate it with images we've built for ourselves?

Reply:
After discussion. We need to keep this list as there is no default setting for kserver deployment (as it is e.g. in k-native). Moreover this list will be needed when we will use custom default images e.g. with rocks.